test/e2e/mcd_test.go

```diff
@@ -425,6 +425,45 @@ func TestReconcileAfterBadMC(t *testing.T) {
 	}
 }
 
+func TestFullyDeleteMCShippedFiles(t *testing.T) {
+	cs := framework.NewClientSet("")
+	nodes, err := getNodesByRole(cs, "worker")
+	require.Nil(t, err)
+	for _, node := range nodes {
+		// this fails if the file is on the node, just a pre-req sanity check to run this test
+		_, err := execCmdOnNodeWithErr(t, cs, node, "test", "!", "-e", "/rootfs/etc/something")
+		require.Nil(t, err)
+	}
+	workerOldMcFirst := getMcName(t, cs, "worker")
+	mcHostFileFirst := createMCToAddFile("modify-host-file-first", "/etc/something", "mco-test", "root")
```

Comment:
I wonder if we could hack it a bit to apply both files in the same MC. When we iterate through the new files it will see both instances as separate. Then we can delete both files in one transaction as well, so we go from 4 reboots to 2.

That said this hack will likely not play nicely with ign spec3.

Reply:
As in, we will see the same functionality being tested whether we update the file through 2 MCs or 1 MC